smartsim/settings/batchSettings.py

```diff
@@ -44,12 +44,73 @@
 
 
 class BatchSettings(BaseSettings):
```

Comment:
Github won't let me comment because it wasn't a modified line, but for teh setter methods on ``env_vars`` shoudl we implement similiar checks that Julia is implementing on the type since at the user-API level we wil rely on runtime checking?